core/src/main/java/org/elasticsearch/search/aggregations/AggregatorBase.java

```diff
@@ -265,9 +280,7 @@ public void close() {
         try {
             doClose();
         } finally {
-            if (!this.failed) {
```

Comment:
>It'd almost be nice to have the changes to this file pulled out into a separate PR.

Makes sense. I found this code a little surprising as in the flag name didn't immediately tell me what failed and how. It wasn't being used anywhere else so I changed it around to be a measure of bytes allocated by the agg + subclasses.